src/main/java/org/cryptomator/linux/quickaccess/DolphinPlaces.java

```diff
@@ -0,0 +1,161 @@
+			xmlValidator = factory.newSchema(schemaFile).newValidator();
+		} catch (IOException | SAXException e) {
+			throw new IllegalStateException("Failed to load included XBEL schema definition file.", e);
+		}
+	}
+
+
+	@Override
+	public QuickAccessService.QuickAccessEntry add(Path target, String displayName) throws QuickAccessServiceException {
+		String id = UUID.randomUUID().toString();
+		try {
+			MODIFY_LOCK.lock();
+			if (Files.size(PLACES_FILE) > MAX_FILE_SIZE) {
+				throw new IOException("File %s exceeds size of %d bytes".formatted(PLACES_FILE, MAX_FILE_SIZE));
+			}
+			var placesContent = Files.readString(PLACES_FILE);
+			//validate
+			xmlValidator.validate(new StreamSource(new StringReader(placesContent)));
+			// modify
+			int insertIndex = placesContent.lastIndexOf("</xbel>"); //cannot be -1 due to validation
+			try (var writer = Files.newBufferedWriter(TMP_FILE, StandardCharsets.UTF_8, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+				writer.write(placesContent, 0, insertIndex);
+				writer.newLine();
+				writer.write(ENTRY_TEMPLATE.formatted(target.toUri(), displayName, id).indent(1));
+				writer.newLine();
+				writer.write(placesContent, insertIndex, placesContent.length() - insertIndex);
+			}
+			// save
+			Files.move(TMP_FILE, PLACES_FILE, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+			return new DolphinPlacesEntry(id);
+		} catch (SAXException | IOException e) {
+			throw new QuickAccessServiceException("Adding entry to KDE places file failed.", e);
+		} finally {
+			MODIFY_LOCK.unlock();
+		}
+	}
+
+	private static class DolphinPlacesEntry implements QuickAccessEntry {
+
+		private final String id;
+		private volatile boolean isRemoved = false;
+
+		DolphinPlacesEntry(String id) {
+			this.id = id;
+		}
+
+		@Override
+		public void remove() throws QuickAccessServiceException {
+			try {
+				MODIFY_LOCK.lock();
+				if (isRemoved) {
+					return;
+				}
+				if (Files.size(PLACES_FILE) > MAX_FILE_SIZE) {
+					throw new IOException("File %s exceeds size of %d bytes".formatted(PLACES_FILE, MAX_FILE_SIZE));
+				}
+				var placesContent = Files.readString(PLACES_FILE);
+				int idIndex = placesContent.lastIndexOf(id);
+				if (idIndex == -1) {
+					isRemoved = true;
+					return; //we assume someone has removed our entry
+				}
+				//validate
+				xmlValidator.validate(new StreamSource(new StringReader(placesContent)));
+				//modify
+				var placesContentPart1 = placesContent.substring(0, idIndex);
+				int openingTagIndex = placesContentPart1.lastIndexOf("<bookmark href=");
+				var contentToWrite1 = placesContentPart1.substring(0, openingTagIndex).stripTrailing();
+
+				int closingTagIndex = placesContent.indexOf("</bookmark>", idIndex);
+				var part2Tmp = placesContent.substring(closingTagIndex + "</bookmark>".length()).split("\\v*", 2); //removing leading vertical whitespaces
```

Comment:
This split won't work as expected. As an example with `\\s*` instead of `\\v*`:
```
" Hello world".split("\\s*", 2) // { "", "Hello world" } -- Expected result
"Hello world".split("\\s*", 2) // { "H", "ello world" } -- Unexpected result
```

Reply:
Fixed in https://github.com/cryptomator/integrations-linux/pull/85/commits/5e9eacc5cbd51c4dc3b1ea8a9d5734ad591c49fb